core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3611,12 +3611,17 @@ class KafkaApis(val requestChannel: RequestChannel,
       clusterId,
       () => {
         val brokers = new DescribeClusterResponseData.DescribeClusterBrokerCollection()
-        metadataCache.getAliveBrokerNodes(request.context.listenerName).foreach { node =>
+        val describeClusterRequest = request.body[DescribeClusterRequest]
+        metadataCache.getBrokerNodes(request.context.listenerName).foreach { node =>
+          if (node.isFenced && !describeClusterRequest.data().includeFencedBrokers()) {
+          } else {
```

Comment:
What do we want to do in the if block? log something? Or remove if / else?

Reply:
I removed the else and swapped the condition around